### PR TITLE
Directional, point and cone lights

### PR DESCRIPTION
**Pull request by MrPepperoni**

### CODE REVIEW

**Comment by MrPepperoni** [comment]
don't merge yet, i fucked up something, seems to leak like mad

**Comment by MrPepperoni** [comment]
Leak fixed with b9f8a84

File: run/data/shaders/postprocessor_select.vs [hunk 1]
```diff
@@ -16,7 +16,8 @@ layout(location=4) in vec2 Size;
 layout(location=5) in vec4 color;
 layout(location=6) in vec4 procTexCoord;
 layout(location=7) in vec4 normalTexCoord;
-layout(location=8) in vec4 proccolor;
+layout(location=8) in float ShadowLenScale;
```
Comment (HalalUr) [inline run/data/shaders/postprocessor_select.vs:19]: shadowLenScale

File: run/data/shaders/postprocessor_select.vs [hunk 2]
```diff
@@ -16,7 +16,8 @@ layout(location=4) in vec2 Size;
...
+layout(location=9) in vec4 proccolor;
```
Comment (HalalUr) [inline run/data/shaders/postprocessor_select.vs:20]: procColor

File: run/data/shaders/lights.vs [hunk 3]
```diff
@@ -0,0 +1,18 @@
...
+//---global uniforms
+layout(std140) uniform GlobalMatrices
+{
+    mat4 cameraToClipMatrix;
+    mat4 worldToCameraMatrix;
+};
+
+//---shader spec
+layout(location=0) in vec2 position;
+layout(location=1) in vec2 texcoord;
```
Comment (HalalUr) [inline run/data/shaders/lights.vs:12]: texCoord

File: src/render/actor_renderer.cpp [hunk 4]
```diff
@@ -178,7 +188,7 @@ bool getNextTextId( RenderableSprites_t::const_iterator& i, RenderableSprites_t:
 }
 }
 
-void ActorRenderer::Prepare( Scene const& , Camera const& camera, double DeltaTime )
+void ActorRenderer::Prepare( Scene const& , std::vector<Camera const*> const& cameras, double DeltaTime )
```
Comment (HalalUr) [inline src/render/actor_renderer.cpp:191]: I cannot find the place where it is called. (it should be in this pr is't it?)

**Review by HalalUr — APPROVED** [review]
I approve this. the naming in shaders is not a real thing, i just mentioned them. The abscence of the prepare call is a bit strange for me with these new cameras vector. But it I trust you man. +3.200 lines of code can't broke a thing!

**Pull request merged**